### PR TITLE
Adds IOT samples

### PR DESCRIPTION
**Pull request by bshaffer**
This is still waiting on changes to gax and google/cloud, but otherwise it's ready for review!

### CODE REVIEW

File: iot/composer.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            "src/list_registries.php",
+            "src/patch_es.php",
+            "src/patch_rsa.php",
+            "src/set_device_config.php",
+            "src/set_device_state.php",
+            "src/set_iam_policy.php"
+        ]
+    },
+    "require-dev": {
+        "phpunit/phpunit": "~4"
```
[the remark this answers is not in the document]
Comment (tmatsuo) [reply iot/composer.json]: I thought I started to use newer phpunit for some tests. So I think there are some tests which uses newer phpunit.

[the remark this answers is not in the document]
Comment (tmatsuo) [reply iot/composer.json]: I agree that we should upgrade the rest, but it is also true that adding a new test with phpunit 4 is not the best practice. I would use newer phpunit for new tests.

[the remark this answers is not in the document]
Comment (bshaffer) [reply iot/composer.json]: I am okay with moving to a newer version, but looking in this repository I get the following PHPUnit versions:

```sh
find . -name composer.json -not -path '*/vendor/*'  | xargs cat  | grep phpunit 
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4.8",
        "phpunit/phpunit": "^4.0"
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4",
        "phpunit/phpunit":"~4.8",
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4",
        "phpunit/phpunit": "~4.8",
        "phpunit/phpunit": "~4",
        "phpunit/phpunit": "~4.0"
        "phpunit/phpunit": "~4.8"
        "phpunit/phpunit": "~4.8"
        "phpunit/phpunit": "~4.8"
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4.8"
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4.8"
        "phpunit/phpunit": "~4.8",
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4"
        "phpunit/phpunit": "~4.8"
        "phpunit/phpunit": "~4.8",
        "phpunit/phpunit": "^4"
        "phpunit/phpunit": "~4.8",
        "phpunit/phpunit": "~4.8",
```

[the remark this answers is not in the document]
Comment (bshaffer) [reply iot/composer.json]: I've upgraded to `phpunit/phpunit:^5` and it works great! PHPunit 6 is not compatible with PHP 5.6, so we'll have to wait before upgrading any further.

[the remark this answers is not in the document]
Comment (tmatsuo) [reply iot/composer.json]: Thx! I'll let @ryanmats  review the code :)

File: iot/src/create_registry.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+namespace Google\Cloud\Samples\Iot;
+
+# [START iot_create_registry]
+use Google\Cloud\Iot\V1\DeviceManagerClient;
+use Google\Cloud\Iot\V1\DeviceRegistry;
+use Google\Cloud\Iot\V1\EventNotificationConfig;
+
+/**
+ * Creates a registry and returns the result. Returns an empty result if
+ * the registry already exists.
```
Comment (ryanmats) [inline iot/src/create_registry.php]: The comment says it's returning the result but it doesn't look like it's returning anything. Also, a bit confused about what happens if the registry already exists. Would $registry be empty or would it contain information (Id/Name) from the already existing registry?

Comment (bshaffer) [reply iot/src/create_registry.php]: I'll update this because you're right, it's incorrect. It throws an error saying the registry exists.

**Pull request merged**